Fxr preview bugfixes pmo

patrickoshaughnessey opened the pull request:
Add 2D projection mode button to projection menu
Fix stereo side-by-side distance from viewer

patrickoshaughnessey committed:
Add 2D projection with developer-art button

patrickoshaughnessey committed:
Uncomment the 3d side-by-side transform matrix, fixing the re-declaration issue

thomasmo committed:
Update and add 2D projection dev icon

patrickoshaughnessey committed:
Add designer provided icon for 2D projection mode

patrickoshaughnessey committed:
Merge branch 'fxr_preview_bugfixes_pmo' of https://github.com/MozillaReality/gecko-dev into fxr_preview_bugfixes_pmo

# Conflicts:
#	browser/fxr/content/assets/icon-2d.svg

patrickoshaughnessey committed:
Merge branch 'fxr_preview_bugfixes' into fxr_preview_bugfixes_pmo

patrickoshaughnessey committed:
Add logic to grab the full screen hint from the URI query string and pass it into the window manager
Remove the automatic switch to 360 projection mode unless we get a hint in the URL

thomasmo commented on gfx/vr/FxRWindowManager.cpp:
```diff
@@ -613,15 +613,14 @@ void FxRWindowManager::OnFullScreenChange(uint64_t aOuterWindowID,
+	  MOZ_ASSERT(overlayError == vr::VROverlayError_None);
     }
 
     MOZ_ASSERT(overlayError == vr::VROverlayError_None);
```
this should have a build failure on a debug build. please remove this line

patrickoshaughnessey replied:
Line 623?

thomasmo replied:
not 623, but 627 (since the scope of overlayError is now inside that else block)
MOZ_ASSERT should only be defined in debug, not release, which is why you wouldn't see that compile failure

patrickoshaughnessey replied:
Ah, cool, got it. thanks

thomasmo commented on browser/fxr/content/assets/icon-2d.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 200 200"><g id="Layer_2" data-name="Layer 2"><path d="M184.9,24.29H15.1a8,8,0,0,0-8,8V167.71a8,8,0,0,0,8,8H184.9a8,8,0,0,0,8-8V32.29A8,8,0,0,0,184.9,24.29ZM23.1,159.71V40.29H176.9V159.71Z"/></g></svg>
```
two things:
1) remove the attributes from <g> (i.e., id, data-name)
2) add the fill attribute to <svg>, i.e.
`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 200 200" fill="context-fill">`

thomasmo replied:
also, please add the license header to the beginnign of the file (my bad for not doing that earlier)
```
<!-- This Source Code Form is subject to the terms of the Mozilla Public
   - License, v. 2.0. If a copy of the MPL was not distributed with this
   - file, You can obtain one at https://mozilla.org/MPL/2.0/. -->
```

thomasmo commented on browser/fxr/content/fxr-fullScreen.js:
```diff
@@ -42,6 +42,38 @@ var FullScreen = {
+    // Get the mozVideoProjection query string if there is one, and use it to set the projection mode for FxR
+    // See: https://github.com/MozillaReality/FirefoxReality/wiki/Media-Playback
+    // NOTE: Currently "Stereo 180 Left to Right" and "Stereo 180 Top to Bottom" are unsupported
+    var currentFullscreenURI = document.fullscreenElement.currentURI;
```
please use `let` rather than `var` to limit the scope

thomasmo commented on browser/fxr/content/fxr-fullScreen.js:
```diff
@@ -42,6 +42,38 @@ var FullScreen = {
+      var mozVideoProjectionIndex = currentFullscreenURI.query.indexOf("mozVideoProjection=");
+      if (mozVideoProjectionIndex != -1)
+      {
+        var projectionMode = currentFullscreenURI.query.substring(currentFullscreenURI.query.indexOf("mozVideoProjection=") + 19);
```
a few things: either
make `mozVideoProjection=` a `const` variable, and then use its length property (instead of magic number 19 and repeated string)
or
use URLSearchParams to get the value

patrickoshaughnessey committed:
Updates based on code review
Changed var to let to reduce scope
Changed parsing to use URLSearchParams class
Added copy right notice to icon-2d.svg, and add teh fill="content-fill" attribute, so it snows up white instead of black
Removed extraneous MOZ_ASSERT(overlayError == vr::VROverlayError_None);

patrickoshaughnessey committed:
Fix getting projection mode from the query parameters - remove extraneous "=" in query parameter

patrickoshaughnessey committed:
make "mozVideoProjection" string literal a const
